src/app/SERVICE/tasks.service.ts

```diff
@@ -1,48 +1,50 @@
 import { Injectable } from '@angular/core';
 import { Task } from '../models/task.model';
-import { BehaviorSubject, Observable, map } from 'rxjs';
+import { Observable, map, switchMap } from 'rxjs';
 import { HttpClient } from '@angular/common/http';
 import { Router } from '@angular/router';
-import { Constants } from '../utils/constants';
+import { tasksApiUrl } from '../utils/constants';
 
 @Injectable({
   providedIn: 'root'
 })
 export class TasksService {
 
-  constructor(private http: HttpClient, private router: Router, private constants: Constants) {
+  constructor(private http: HttpClient, private router: Router) {
   }
 
-  public getAllTasks():Observable<Task[]>{
-    return this.http.get<Task[]>(this.constants.tasksApiUrl)
+  getAllTasks(): Observable<Task[]> {
+    return this.http.get<Task[]>(tasksApiUrl)
       .pipe(
-        map(data => { return data })
+        map(data => data)
       );
   }
 
-  async createNewTask(task: Task) {
+  createNewTask(task: Task) {
     if (task) {
-      await this.http.post(this.constants.tasksApiUrl, task).subscribe()
+      this.http.post(tasksApiUrl, task).subscribe()
```

Comment:
remove subscribe and find a different solution as I said having subscriptions in services will cause memory leaks as you can't really unsubscribe from them easily as you can in the components